comp/core/workloadmeta/collectors/internal/containerd/image_sbom_trivy.go

```diff
@@ -76,46 +72,7 @@ func (c *collector) startSBOMCollection(ctx context.Context) error {
 		}
 	}()
 
-	go func() {
-		for result := range resultChan {
-			if result.ImgMeta == nil {
-				log.Errorf("Scan result does not hold the image identifier. Error: %s", result.Error)
-				continue
-			}
-
-			status := workloadmeta.Success
-			reportedError := ""
-			var report *cyclonedx.BOM
-			if result.Error != nil {
-				// TODO: add a retry mechanism for retryable errors
-				log.Errorf("Failed to generate SBOM for containerd image: %s", result.Error)
-				status = workloadmeta.Failed
-				reportedError = result.Error.Error()
-			} else {
-				bom, err := result.Report.ToCycloneDX()
-				if err != nil {
-					log.Errorf("Failed to extract SBOM from report")
-					status = workloadmeta.Failed
-					reportedError = err.Error()
-				}
-				report = bom
-			}
-
-			sbom := &workloadmeta.SBOM{
-				CycloneDXBOM:       report,
-				GenerationTime:     result.CreatedAt,
-				GenerationDuration: result.Duration,
-				Status:             status,
-				Error:              reportedError,
-			}
-
-			// Updating workloadmeta entities directly is not thread-safe, that's why we
-			// generate an update event here instead.
-			if err := c.handleImageCreateOrUpdate(ctx, result.ImgMeta.Namespace, result.ImgMeta.Name, sbom); err != nil {
-				log.Warnf("Error extracting SBOM for image: namespace=%s name=%s, err: %s", result.ImgMeta.Namespace, result.ImgMeta.Name, err)
-			}
-		}
-	}()
+	go c.startScanResultHandler(ctx, resultChan)
```

Comment:
to manage a clean stop you can add a `sync.WaitGroup` and wait that  `startScanResultHandler()` stop properly

Reply:
I think that by doing so we would change the behavior of the `startSBOMCollection` function which should start goroutines and terminate instantly, causing possible deadlocks. IMO as long as no state is shared between two goroutines, maybe we can keep them independent ?